src/category_theory/limits/shapes/constructions/equalizers.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import category_theory.limits.shapes
+
+/-!
+# Constructing equalizers from pullbacks and binary products.
+
+If a category has pullbacks and binary products, then it has equalizers.
+
+TODO: provide the dual result.
+-/
+universes v u
+
+open category_theory category_theory.category category_theory.limits
+
+variables {C : Type u} [𝒞 : category.{v} C] [has_binary_products.{v} C] [has_pullbacks.{v} C]
+include 𝒞
+
+@[reducible]
+def construct_equalizer (F : walking_parallel_pair ⥤ C) : C :=
+pullback (prod.lift (𝟙 _) (F.map walking_parallel_pair_hom.left))
+         (prod.lift (𝟙 _) (F.map walking_parallel_pair_hom.right))
+
+@[reducible]
+def construct_ι (F : walking_parallel_pair ⥤ C) :
+  construct_equalizer F ⟶ F.obj walking_parallel_pair.zero :=
+pullback.fst
+
+lemma construct_ι_eq_snd (F : walking_parallel_pair ⥤ C) : construct_ι F = pullback.snd :=
+begin
+  have l: (pullback.fst ≫ prod.lift (𝟙 _) (F.map walking_parallel_pair_hom.left)) ≫ limits.prod.fst = (pullback.snd ≫ prod.lift (𝟙 _) (F.map walking_parallel_pair_hom.right)) ≫ limits.prod.fst,
+    rw pullback.condition,
+  erw [assoc, assoc, limit.lift_π, limit.lift_π, comp_id, comp_id] at l, exact l
+end
+
+@[reducible]
+def equalizer_cone (F : walking_parallel_pair ⥤ C) : cone F :=
+cone.of_fork
+  (fork.of_ι (construct_ι F)
+    (begin
+      have r: (pullback.fst ≫ prod.lift (𝟙 _) (F.map walking_parallel_pair_hom.left)) ≫ limits.prod.snd = (pullback.snd ≫ prod.lift (𝟙 _) (F.map walking_parallel_pair_hom.right)) ≫ limits.prod.snd,
+        rw pullback.condition,
+      simp only [limit.lift_π, assoc] at r,
+      erw r, rw construct_ι_eq_snd, refl
+     end))
+
+def equalizer_cone_is_limit (F : walking_parallel_pair ⥤ C) : is_limit (equalizer_cone F) :=
+{ lift :=
+  begin
+    intro c, apply pullback.lift (c.π.app _) (c.π.app _),
+    apply limit.hom_ext,
+    rintro (_ | _), all_goals { simp [assoc, limit.lift_π] }
+  end,
+  fac' :=
+  begin
+    intro c, rintro (_ | _),
+    { simp, refl },
+    { simp, exact c.w _ }
+  end,
+  uniq' :=
+  begin
+    intros c _ J,
+    have J1 := J walking_parallel_pair.zero, simp at J1,
+    apply pullback.hom_ext,
+    { rwa limit.lift_π },
+    { erw [limit.lift_π, ← J1, construct_ι_eq_snd] }
+  end
+}
+
+def has_equalizers_of_pullbacks_and_binary_products
```

Comment:
When you add a doc-string, explicitly say that you are not turning this into an instance.

Reply:
Edited, thanks!